geomesa-blobstore/geomesa-blobstore-api/src/main/scala/org/locationtech/geomesa/blob/api/BlobstoreServlet.scala

```diff
@@ -8,55 +8,240 @@
     val ds = new AccumuloDataStoreFactory().createDataStore(dsParams).asInstanceOf[AccumuloDataStore]
+    if (ds == null) {
+      logger.error("Bad Connection Params: {}", dsParams)
+      None
+    } else {
+      Some(new AccumuloBlobStore(ds))
+    }
+  }
+
+  /**
+    * Override of the requestPath function, may need to match on type of request if wrapped
+    * depending on deployment environment
+    *
+    * @param request
+    * @return
+    */
+  override def requestPath(implicit request: HttpServletRequest): String = request match {
+    case r: HttpServletRequestWrapper =>
+      r.getMethod.toLowerCase(Locale.US) match {
+        case "post" =>
+          if (r.getServletPath == "/blob") {
+            s"/blob${r.getPathInfo}"
+          } else {
+            r.getPathInfo
+          }
+        case "delete" =>
+          r.getPathInfo.replace(s"/$root", "")
+        case _        =>
+          r.getPathInfo
+      }
+    case _ => super.requestPath
+  }
+
+  private def handleBadRequest(msg: String): ActionResult = {
+    logger.error(msg)
+    if (debug) {
+      BadRequest(reason = msg)
+    } else {
+      BadRequest()
+    }
+  }
 
+  // TODO: Revisit configuration and persistence of configuration.
+  // https://geomesa.atlassian.net/browse/GEOMESA-958
+  /**
+    * Registers a data store, making it available for later use
+    */
+  post("/ds/:alias/?") {
+    logger.debug("Attempting to register accumulo connection in Blob Store")
+    val dsParams = datastoreParams
+    val ds = new AccumuloDataStoreFactory().createDataStore(dsParams).asInstanceOf[AccumuloDataStore]
     if (ds == null) {
-      NotFound(reason = "Could not load data store using the provided parameters.")
+      handleBadRequest("Could not load data store using the provided parameters.")
     } else {
-      // TODO: Synchronize Blobstore creation
-      // https://geomesa.atlassian.net/browse/GEOMESA-985
-      abs = new AccumuloBlobStore(ds)
+      val alias = params("alias")
+      val prefix = keyFor(alias)
+      val toPersist = dsParams.map { case (k, v) => keyFor(alias, k) -> v }
+      try {
+        persistence.removeAll(persistence.keys(prefix).toSeq)
+        persistence.persistAll(toPersist)
+        blobStores.put(alias, new AccumuloBlobStore(ds))
+        Ok()
+      } catch {
+        case e: Exception => handleError(s"Error persisting data store '$alias':", e)
+      }
+    }
+  }
+
+  /**
+    * Retrieve an existing data store
+    */
+  get("/ds/:alias/?") {
+    try {
+      getPersistedDataStore(params("alias"))
+    } catch {
+      case e: Exception => handleError(s"Error reading data store:", e)
+    }
+  }
+
+  /**
+    * Remove the reference to an existing data store and removes tables
+    */
+  delete("/ds/:alias/?") {
+    val alias = params("alias")
+    val prefix = keyFor(alias)
+    try {
+      blobStores.get(alias).foreach(_.delete())
```

Comment:
is this deleting actual accumulo data? i think you just want to remove reference from the map.